src/components/InventoryDetail/DetailHeader.js

```diff
@@ -0,0 +1,96 @@
+import React, { Fragment } from 'react';
+import PropTypes from 'prop-types';
+import './InventoryDetail.scss';
+import TopBar from './TopBar';
+import FactsInfo from './FactsInfo';
+import './InventoryDetail.scss';
+import InsightsPrompt from './InsightsPrompt';
+import { verifyCulledInsightsClient } from '../../Utilities/sharedFunctions';
+import { PageHeader } from '@redhat-cloud-services/frontend-components';
+import classnames from 'classnames';
+
+const DetailHeader = ({
+    entity,
+    hideInvLink,
+    loaded,
+    addNotification,
+    deleteEntity,
+    onBackToListClick,
+    actions,
+    UUIDWrapper,
+    LastSeenWrapper,
+    children,
+    showInventoryDrawer,
+    shouldWrapAsPage,
+    BreadcrumbWrapper,
+    additionalClasses,
+    showDelete,
+    showTags,
+    TitleWrapper,
+    TagsWrapper,
+    DeleteWrapper,
+    ActionsWrapper
+}) => {
+    const HeaderInfo = (<Fragment>
+        <TopBar
+            entity={entity}
+            loaded={loaded}
+            onBackToListClick={onBackToListClick}
+            actions={actions}
+            deleteEntity={deleteEntity}
+            addNotification={addNotification}
+            hideInvLink={hideInvLink}
+            showInventoryDrawer={showInventoryDrawer}
+            showDelete={showDelete}
+            showTags={showTags}
+            TitleWrapper={TitleWrapper}
+            TagsWrapper={TagsWrapper}
+            DeleteWrapper={DeleteWrapper}
+            ActionsWrapper={ActionsWrapper}
+        />
+        <FactsInfo
+            loaded={loaded}
+            entity={entity}
+            UUIDWrapper={UUIDWrapper}
+            LastSeenWrapper={LastSeenWrapper}
+        />
+        {(loaded && verifyCulledInsightsClient(entity?.per_reporter_staleness)) && <InsightsPrompt />}
+        {children}
+    </Fragment>);
+
+    return (shouldWrapAsPage ?
+        (<PageHeader className={classnames('pf-m-light ins-inventory-detail', additionalClasses)} >
+            {BreadcrumbWrapper}
+            {HeaderInfo}
+        </PageHeader>) : HeaderInfo);
+};
```

Comment:
It's not a good idea to create new component on each render (especially with so many props). How about we'd split it into two components?
```suggestion

const HeaderInfo = ({ entity, loaded, UUIDWrapper, LastSeenWrapper, children, ...props }) =>  (<Fragment>
        <TopBar
            entity={entity}
            loaded={loaded}
            {...props}
        />
        <FactsInfo
            loaded={loaded}
            entity={entity}
            UUIDWrapper={UUIDWrapper}
            LastSeenWrapper={LastSeenWrapper}
        />
        {(loaded && verifyCulledInsightsClient(entity?.per_reporter_staleness)) && <InsightsPrompt />}
        {children}
    </Fragment>);

const DetailHeader = ({ shouldWrapAsPage, ...props }) => {
    return (shouldWrapAsPage ?
        (<PageHeader className={classnames('pf-m-light ins-inventory-detail', additionalClasses)} >
            {BreadcrumbWrapper}
            <HeaderInfo {...props} />
        </PageHeader>) : <HeaderInfo {...props} />);
};

HeaderInfo.propTypes = DetailHeader.propTypes;
```

I am not sure if I didn't missaligned anything, better to try this locally.

Reply:
Of course. I missed this caveat. I will adopt the suggestion